franklin/github/views.py

```diff
@@ -63,16 +217,30 @@ def deploy_hook(request):
         # Possibly useful for validation/security 
         # request.META.get("HTTP_X_GITHUB_DELIVERY")
 
-        if event_type and event_type == "push":
-            # For now, we only support push to branch events
-            # TODO - We should also only support pushes to specific branches
-
-            updated_site = GithubWebhookSerializer(data=request.data)
-            if updated_site.is_valid():
-                updated_site.save()
-                return Response(status=status.HTTP_201_CREATED)
-            else:
-                logger.warning("Received an invalid Github Webhook message")
+        if event_type:
+            if event_type == 'push':
+                # For now, we only support push to branch events
+                # TODO - We should also only support pushes to specific branches
+
+                github_event = GithubWebhookSerializer(data=request.data)
+                if github_event:
+                    site = github_event.get_existing_site()
+                    if site and site.is_deployable_event(github_event):
+                        site.save()
+                        # This line helps with testing. We will remove once we add mocking.
+                        if os.environ['ENV'] is not 'test':
+                            site.build()
+                            return Response(status=status.HTTP_201_CREATED)
+                else:
+                    logger.warning("Received an invalid Github Webhook message")
+            elif event_type == 'ping':
+                # TODO - update the DB with some important info here
+                # repository{ 
+                #           id, name, 
+                #           owner{ id, login }, 
+                #           sender{ id, login, site_admin }
+                #           }
```

Comment:
This is the webhook event github calls with to make sure our endpoint is valid. They send us some good detail on the project at this point. Saving this data here will mean less we have to troll for when we build the `/admin` page.